packages/runtime/src/services/StateManager.ts

```diff
@@ -157,58 +164,67 @@ export class StateManager {
     }) as PropsAfterEvaled<T>;
   }
 
-  deepEval<T extends Record<string, unknown> | any[]>(
-    obj: T,
+  deepEval<T extends Record<string, unknown> | any[] | string>(
+    value: T,
     options: EvalOptions = {}
-  ): PropsAfterEvaled<T> {
+  ): T extends string ? unknown : PropsAfterEvaled<Exclude<T, string>> {
     // just eval
-    const evaluated = this.mapValuesDeep(obj, ({ value }) => {
-      if (typeof value !== 'string') {
-        return value;
-      }
-      return this.maskedEval(value, options);
-    });
-
-    return evaluated;
+    if (typeof value !== 'string') {
+      return this.mapValuesDeep(value, ({ value }) => {
+        if (typeof value !== 'string') {
+          return value;
+        }
+        return this._maskedEval(value, options);
+      }) as T extends string ? unknown : PropsAfterEvaled<Exclude<T, string>>;
+    } else {
+      return this._maskedEval(value, options) as T extends string
+        ? unknown
+        : PropsAfterEvaled<Exclude<T, string>>;
+    }
   }
 
-  deepEvalAndWatch<T extends Record<string, unknown> | any[]>(
-    obj: T,
-    watcher: (params: { result: PropsAfterEvaled<T> }) => void,
+  deepEvalAndWatch<T extends Record<string, unknown> | any[] | string>(
+    value: T,
+    watcher: (params: { result: PropsAfterEvaled<Exclude<T, string>> }) => void,
     options: EvalOptions = {}
   ) {
     const stops: ReturnType<typeof watch>[] = [];
 
     // just eval
-    const evaluated = this.deepEval(obj, options);
-
-    // watch change
-    let resultCache: PropsAfterEvaled<T> = evaluated;
-    this.mapValuesDeep(obj, ({ value, path }) => {
-      const isDynamicExpression =
-        typeof value === 'string' &&
-        parseExpression(value).some(exp => typeof exp !== 'string');
-
-      if (!isDynamicExpression) return;
-
-      const stop = watch(
-        () => {
-          const result = this.maskedEval(value as string, options);
-
-          return result;
-        },
-        newV => {
-          if (isProxy(newV)) {
-            newV = toRaw(newV);
+    const evaluated = this.deepEval(value, options) as T extends string
+      ? unknown
+      : PropsAfterEvaled<Exclude<T, string>>;
+
+    if (value && typeof value === 'object') {
```

Comment:
If value is string, I think maybe we also need to watch it?

Reply:
Done.